warehouse/index.js

```diff
@@ -173,26 +173,32 @@ function setDataFromInputAndComputeColumnTypes(
   input,
   columnTypes,
   options,
-  prefix = ""
+  prefix = "",
+  level = 0
 ) {
   if (!input || !isObject(input)) return;
   Object.keys(input).forEach(key => {
-    if (isObject(input[key])) {
+    if (isObject(input[key]) && (options.sourceCategory == 'cloud' && level < 3)) {
```

Comment:
Should it be something like
`isObject(input[key]) && (options.sourceCategory !== 'cloud' || level < 3)`

Reply:
Let us add couple of test cases in `warehouse.test.js`